3rdparty/mshadow/mshadow/tensor.h

```diff
@@ -89,15 +89,33 @@ struct Shape {
    * \return the corresponding dimension size
    */
   MSHADOW_XINLINE index_t &operator[](int idx) {
+// two possible solutions: throw exception if idx is out of bounds
+//                         --> costs runtime performance
+//                         disable warning and let the user take care
+//                         --> UB possible if user doesn't take care
+#pragma GCC diagnostic push
+#if __GNUC__ >= 8
+#pragma GCC diagnostic ignored "-Warray-bounds"
```

Comment:
Warray-bounds warns about subscripts to arrays that are always out of bounds. Is there any evidence for GCC being wrong and the subscript being in bounds?

Reply:
My understanding was that it checks out-of-range access at runtime, it seams that I'm wrong.
But then there is a possible flaw in MXNet, because the compiler has had detected an out-of-range access at compile-time (meaning from the MXNet sources itself).

Reply:
It's either a compiler bug or MXNet bug ;) Did you take a look at the source file that triggers the warning?